src/test/java/spoon/test/comment/CommentTest.java

```diff
@@ -775,6 +776,78 @@ public void testSnippedWithComments() {
 		assertEquals("inline comment", returnSt.getComments().get(1).getContent());
 	}
 
+	@Test
+	public void testEqualsWithDifferentClassObjects() {
+		// contract: equals return false when two objects of different classes are compared
+
+		Factory factory = new Launcher().getFactory();
+		CtNamedElement object = factory.Interface().get(CtNamedElement.class);
+		CtCommentImpl ctComment = new CtCommentImpl();
+
+		Boolean expectedBoolean = ctComment.equals(object);
+
+		assertFalse(expectedBoolean);
+	}
+
+	@Test
+	public void testEqualsWithDifferentContent() {
+		// contract: equals return false when two comments with different contents are compared
+
+		Factory factory = new Launcher().getFactory();
+
+		CtComment comment = factory.Core().createComment();
+		String commentContent = "testContent";
+		comment.setContent(commentContent);
+
+		CtCommentImpl ctComment = new CtCommentImpl();
+		String ctCommentContent = "notTestContent";
+		ctComment.setContent(ctCommentContent);
+
+		boolean expectedBoolean = ctComment.equals(comment);
+
+		assertFalse(expectedBoolean);
+	}
+
+	@Test
+	public void testEqualsWithSameContentAndSameType() {
+		// contract: equals return true when comments with same content and type are compared
+
+		Factory factory = new Launcher().getFactory();
+
+		CtComment comment = factory.Core().createComment();
+		String commentContent = "testContent";
+		comment.setCommentType(CtComment.CommentType.INLINE);
+		comment.setContent(commentContent);
+
+		CtCommentImpl ctComment = new CtCommentImpl();
+		ctComment.setCommentType(CtComment.CommentType.INLINE);
+		ctComment.setContent(commentContent);
+
+		boolean expectedBoolean = ctComment.equals(comment);
+
+		assertTrue(expectedBoolean);
+	}
+
+	@Test
+	public void testEqualsWithSameContentAndDifferentType() {
+		// contract: equals return false when comments with same content but different types are compared
+
+		Factory factory = new Launcher().getFactory();
+
+		CtComment comment = factory.Core().createComment();
+		String commentContent = "testContent";
+		comment.setCommentType(CtComment.CommentType.BLOCK);
+		comment.setContent(commentContent);
```

Comment:
Same comment about the names.

Can be shortened like so:

```suggestion
		CtComment comment = factory.createComment("testContent", CtComment.CommentType.BLOCK);
```